packages/notification-center/src/hooks/use-session.hook.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery } from '@tanstack/react-query';
+
+import { SESSION_QUERY_KEY } from './queryKeys';
+import type { ISession } from '../shared/interfaces';
+import { useNovuContext } from './use-novu-context.hook';
+
+export const useSession = ({ onSuccess }: { onSuccess?: (session: ISession) => void }) => {
+  const { apiService, applicationIdentifier, subscriberId, subscriberHash } = useNovuContext();
+
+  const result = useQuery<ISession, Error, ISession>(
+    [...SESSION_QUERY_KEY, applicationIdentifier, subscriberId, subscriberHash],
+    () => apiService.initializeSession(applicationIdentifier, subscriberId, subscriberHash),
+    {
+      enabled: !!applicationIdentifier && !!subscriberId,
+      cacheTime: Infinity,
+      staleTime: Infinity,
+      onSuccess,
+      onError: (error: Error) => {
+        // eslint-disable-next-line no-console
+        console.error('Failed to initialize the session', error.message);
+      },
+    }
+  );
+
+  return result;
+};
```

Comment:
`useSession` hook will not be exposed publically as it's an internal logic of the `NovuProvider`.
Fetch the new session only when we have a `applicationIdentifier and subscriberId`, and cache it for an infinite time, so it won't be stale and refetched.